src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -575,6 +588,38 @@ public SNIError GetLastError()
         {
             return SNILoadHandle.SingletonInstance.LastError;
         }
+
+        /// <summary>
+        /// Gets the Local db Named pipe data source if the input is a localDB server. 
+        /// </summary>
+        /// <param name="fullServerName">The data source</param>
+        /// <param name="error">Set true when an error occured while getting LocalDB up</param>
+        /// <returns></returns>
+        private string GetLocalDBDataSource(string fullServerName, out bool error)
+        {
+            string localDBConnectionString = null;
+            bool isBadLocalDBDataSource;
+            string localDBInstance = DataSource.GetLocalDBInstance(fullServerName, out isBadLocalDBDataSource);
+
+            if (isBadLocalDBDataSource)
+            {
+                error = true;
+                return null;
+            }
+            else if (!string.IsNullOrEmpty(localDBInstance))
```

Comment:
I can get rid of the `else if` because it's not expected to be reached in case of empty localDBInstance.